crates/ursa-gateway/src/worker/cache/server.rs

```diff
@@ -82,14 +81,11 @@ async fn fetch_and_insert(
             error!("Failed to dispatch Fetch command: {e:?}");
             anyhow!("Failed to dispatch Fetch command")
         })?;
-    let mut body = match rx
-        .await
-        .map_err(|e| {
-            error!("Failed to receive response from resolver: {e:?}");
-            anyhow!("Failed to receive response from resolver")
-        })??
-        .into_parts()
-    {
+    let response = rx.await.map_err(|e| {
+        error!("Failed to receive response from resolver: {e:?}");
+        anyhow!("Failed to receive response from resolver")
+    })??;
+    let mut body = match response.resp.into_parts() {
```

Comment:
This is an interesting behavior, getting the complete data first and then checking if it's larger than the specified limit for transmission, isn't it better to do a pre-check on the size of the requested cid. 

Do correct me if I am wrong on functionality. Maybe there are other better alternatives but a pre-check is the first one that come to mind.

Reply:
if you already have the data, then why not stream it, think at scale too if the file is GBs worth. Also, this assumes a full content-length grab. instead of a byte range or a stream

Reply:
Technically you will never cross the 1GB mark then if all files are chunked and streamed in that manner. Food for thought

Reply:
True but we know the size of the content because it is advertised (#221) to the global indexer so we have that data point before we have all of the data. `Response.size` is a value returned by indexer.

Reply:
which isn't a trusted value. this can be improved in a later PR, but to note.

Reply:
> which isn't a trusted value. this can be improved in a later PR, but to note.

agree, we can create an issue for this and PR later.

Reply:
created #291, to track this issue.